code/modules/power/antimatter/control.dm

```diff
@@ -268,92 +271,117 @@
 	addtimer(CALLBACK(src, .proc/reset_shield_icon_delay), 20)
 
 /obj/machinery/power/am_control_unit/proc/reset_shield_icon_delay()
-	shield_icon_delay = 0
+	shield_icon_delay = FALSE
 
 /obj/machinery/power/am_control_unit/proc/check_core_stability()
-	if(stored_core_stability_delay || linked_cores.len <= 0)
+	if(linked_cores.len <=0)
 		return
-	stored_core_stability_delay = 1
-	stored_core_stability = 0
+	stored_core_stability = FALSE
 	for(var/obj/machinery/am_shielding/AMS in linked_cores)
 		stored_core_stability += AMS.stability
 	stored_core_stability/=linked_cores.len
-	addtimer(CALLBACK(src, .proc/reset_stored_core_stability_delay), 40)
+	switch(stored_core_stability)
+		if(0 to 24)
+			icon_mod="fuck"
+		if(25 to 49)
+			icon_mod="critical"
+		if(50 to INFINITY)
+			icon_mod="on"
+	update_icon()
 
 /obj/machinery/power/am_control_unit/proc/reset_stored_core_stability_delay()

```

Comment:
As far as I can tell, this proc isn't called anywhere anymore, nor is stored_core_stability_delay used anywhere now.